### PR TITLE
Add sponsors

### LINKED ISSUE #29: Add sponsor information to sensor details

### PR DESCRIPTION
**Pull request by rnestler**
Fixes #29

### CODE REVIEW

File: src/Api.elm [hunk 1]
```diff
@@ -135,3 +143,24 @@ loadSensorMeasurements apiToken now sensorId secondsAgo =
...
+                ++ (toString sponsorId)
+
+        request =
+            Http.request
+                { method = "GET"
+                , headers = getHeaders apiToken
+                , url = url
+                , body = Http.emptyBody
+                , expect = Http.expectJson sponsorDecoder
+                , timeout = Just (30 * Time.second)
```
Comment (dbrgn) [inline src/Api.elm]: We could extract this into a module constant.

File: src/Api.elm [hunk 2]
```diff
@@ -135,3 +143,24 @@ loadSensorMeasurements apiToken now sensorId secondsAgo =
...
     in
         Http.send (\res -> (MeasurementsLoaded ( sensorId, res ))) request
+
+
+loadSponsor : String -> Int -> Cmd Msg
+loadSponsor apiToken sponsorId =
+    let
+        url =
+            "https://watertemp-api.coredump.ch/api/sponsors/"
+                ++ (toString sponsorId)
```
Comment (dbrgn) [inline src/Api.elm]: We could extract this into a function:

    getUrl : String -> String
    getUrl path = "https://watertemp-api.coredump.ch/api" ++ path

File: src/Main.elm [hunk 3]
```diff
@@ -170,6 +187,17 @@ update msg model =
 
                         _ ->
                             Cmd.none
+
+                cmdSponsor =
+                    case (Maybe.map .sponsorId selectedSensor) of
+                        Just (Just sponsorId) ->
+                            Api.loadSponsor model.apiToken (Debug.log "loadSponsor" sponsorId)
```
Comment (dbrgn) [inline src/Main.elm]: You can remove the debug call.

File: src/Views.elm [hunk 4]
```diff
@@ -218,12 +219,24 @@ sidebarContents model =
         [ text <| Maybe.withDefault "Details" (Maybe.map .deviceName model.selectedSensor) ]
     , Maybe.withDefault
         (p [] [ text "Klicke auf einen Sensor, um mehr über ihn zu erfahren." ])
-        (Maybe.map (\s -> sensorDescription model.now s) model.selectedSensor)
+        (Maybe.map
+            (\s ->
```
Comment (dbrgn) [inline src/Views.elm]: It would probably be clearer to rename this to `sensor`.

File: src/Views.elm [hunk 5]
```diff
@@ -279,6 +292,27 @@ sensorDescription now sensor =
...
+        , h3 [] [ text "Sponsor" ]
+        , Maybe.withDefault
+            (p
+                [ css [ fontStyle italic ] ]
+                [ text "Sponsor wird geladen..." ]
+            )
+            (Maybe.map
+                (\sponsor ->
+                    div []
+                        [ (p
```
Comment (dbrgn) [inline src/Views.elm]: Unnecessary parentheses

File: src/Views.elm [hunk 6]
```diff
@@ -279,6 +292,27 @@ sensorDescription now sensor =
...
+                            [ css [ fontStyle italic ] ]
+                            [ text sponsor.name ]
+                          )
+                        , (p
```
Comment (dbrgn) [inline src/Views.elm]: Unnecessary parentheses

File: src/Models.elm [hunk 7]
```diff
@@ -3,13 +3,15 @@ module Models exposing (..)
...
 import Time
+import Dict exposing (Dict)
 
 
 type alias Model =
     { route : Route
     , map : Map.Model
     , sensors : List Sensor
     , selectedSensor : Maybe Sensor
+    , sponsors : Dict Int Sponsor
```
[the remark this answers is not in the document]
Comment (dbrgn) [reply src/Models.elm]: Yes, probably a product type: https://robots.thoughtbot.com/lessons-learned-avoiding-primitives-in-elm

**Comment by dbrgn** [comment]
Ping @rnestler :)

File: src/Main.elm (hunk 3 above)
Comment (rnestler) [reply src/Main.elm]: Done.

File: src/Views.elm (hunk 4 above)
Comment (rnestler) [reply src/Views.elm]: Done. But I didn't change it at every occurrence, to avoid bloating the diff.

File: src/Views.elm (hunk 5 above)
Comment (rnestler) [reply src/Views.elm]: Done.

File: src/Views.elm (hunk 6 above)
Comment (rnestler) [reply src/Views.elm]: Done.

File: src/Models.elm (hunk 7 above)
[the remark this answers is not in the document]
Comment (rnestler) [reply src/Models.elm]: I'll leave this for another PR, since it involves changes at several places.

File: src/Api.elm (hunk 1 above)
Comment (rnestler) [reply src/Api.elm]: Done.

File: src/Api.elm [hunk 8]
```diff
@@ -91,11 +99,21 @@ toJsMeasurement measurement =
 -- API REQUESTS
 
 
+apiTimeout : Time.Time
+apiTimeout =
+    (30 * Time.second)
```
Comment (rnestler) [inline src/Api.elm]: @dbrgn I'm not sure if this is what you meant by module constant.

Comment (dbrgn) [reply src/Api.elm]: Yeah I don't think there's another way :)

**Pull request merged**